src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1777,9 +1777,13 @@ java_bytecode_parsert::parse_method_handle(const method_handle_infot &entry)
   const name_and_type_infot &name_and_type =
     ref_entry.get_name_and_type(pool_entry_lambda);
 
+  std::string class_name = class_entry.get_name(pool_entry_lambda);
```

Comment:
would you mind adding an example here what happens here and why this is required?

Reply:
Do the comments below this line make it a bit clearer perhaps? We need to replace dots by dollars (formatting for inner classes) and then slashes by dots (formatting for packages).

Reply:
yes, it's ok, I was just wondering about the `$` in lambda function names, so I wondered whether we are sure here not to replace those.

Reply:
Yes, that's a good point. They are not being replaced here, only the class name is manipulated, the lambda name is appended to the string afterwards.